src/pages/InitMsig.vue

```diff
@@ -214,63 +229,77 @@ export default defineComponent({
-
-						let sender = algosdk.decodeUnsignedTransaction(
-							convertBase64ToUnit8Array(txnBase64)
-						);
-						let fromAddr = algosdk.encodeAddress(sender.from.publicKey);
-						let addr = this.addresses.map((x) => {
-							return x.address;
-						});
-
-						let version = this.version * 1;
-						let threshold = this.threshold * 1;
-
-						const multisigParams = {
-							version: version,
-							threshold: threshold,
-							addrs: addr,
-						};
-						let multisigaddr = algosdk.multisigAddress(multisigParams);
-						if (fromAddr != multisigaddr) {
-							openErrorNotificationWithIcon(WRONG_ADDRESS, WRONG_ADDRESSES);
+				if (this.walletStore.walletKind) {
+					assertAddrPartOfMultisig(
+						this.msigAddresses,
+						this.walletStore.address
+					);
+					switch (this.walletStore.walletKind) {
+						case WalletType.MY_ALGO: {
+							const { base64, json } = await signMultisigUsingMyAlgoWallet(
+								txnBase64
+							);
+							this.contentList.MSG_PACK = base64;
+							this.contentList.JSON = formatJSON(prettifyTransaction(json));
+							this.key = "JSON";
+							openSuccessNotificationWithIcon(SIGN_SUCCESSFUL);
+							break;
+						}
+						case WalletType.ALGOSIGNER: {
+							let signAlgoSigner = this.walletStore.webMode as WebMode;
+							const multisigParams = this.createMultisig();
+							let signedTxn = await signAlgoSigner.signTransaction([
+								{
+									txn: txnBase64,
+									msig: multisigParams,
+								},
+							]);
+							const json = signedTxn[0] as JsonPayload;
+							this.contentList.MSG_PACK = json.blob as string;
+							const arr = convertBase64ToUint8Array(this.contentList.MSG_PACK);
+							const newJson = algosdk.decodeSignedTransaction(arr);
+							this.contentList.JSON = formatJSON(prettifyTransaction(newJson));
+							openSuccessNotificationWithIcon(SIGN_SUCCESSFUL);
+							this.key = "JSON";
+							break;
+						}
+						case WalletType.WALLET_CONNECT: {
+							openErrorNotificationWithIcon(WALLET_NOT_SUPPORTED);
+							break;
+						}
+						default: {
+							openErrorNotificationWithIcon(NO_WALLET);
 							break;
 						}
-						let signedTxn = await signAlgoSigner.signTransaction([
-							{
-								txn: txnBase64,
-								msig: multisigParams,
-							},
-						]);
-						const json = signedTxn[0] as JsonPayload;
-						this.contentList.MSG_PACK = json.blob as string;
-						const arr = convertBase64ToUnit8Array(this.contentList.MSG_PACK);
-						const newJson = algosdk.decodeSignedTransaction(arr);
-						this.contentList.JSON = formatJSON(prettifyTransaction(newJson));
-						openSuccessNotificationWithIcon(SIGN_SUCCESSFUL);
-						this.key = "JSON";
-						break;
-					}
-					case WalletType.WALLET_CONNECT: {
-						openErrorNotificationWithIcon(NOT_SUPPORT_WALLET);
-						break;
-					}
-					default: {
-						openErrorNotificationWithIcon(NO_WALLET);
-						break;
 					}
-				}
+				} else throw Error("Please connect your wallet.");
```

Comment:
Better put this one into constants

Reply:
It in is constant, I have added it in second PR